### PR TITLE
Api 4660 oauth proxy slugs

### PR DESCRIPTION
**Pull request by crolarlibertyva**
- Currently rewrites idp slugs to its target id based on the configuration.
- Later can be expanded to do fuller rewrites using additional attributes for a given id.
- Retains default behavior is configuration does not contain any idps.

### CODE REVIEW

File: oauth-proxy/dev-config.base.json [hunk 1]
```diff
@@ -57,6 +57,29 @@
...
+      },
+      {
+        "slug": "ssoi",
+        "id": "0uahh383xssoixxxxxx",
+        "manage": "some_manage_endpoint"
+      },
+      {
+        "slug": "ssoe",
+        "id": "0uahxxssoexxxxxxx",
+        "attribute3": "potentially expandable attributes",
```
Comment (beaugrantham-va) [inline oauth-proxy/dev-config.base.json:78]: Let's remove the placeholder attributes until we actually define them.

File: oauth-proxy/dev-config.base.json [hunk 2]
```diff
@@ -57,6 +57,29 @@
       "jwks": "/keys",
       "grants": "/grants",
       "smart_launch": "/smart/launch"
-    }
+    },
+    "idps": [
```
Comment (beaugrantham-va) [inline oauth-proxy/dev-config.base.json:61]: This location (`config.routes.idps`) does not match the slug_helper's expected location (`config.idps`).

Comment (beaugrantham-va) [inline oauth-proxy/dev-config.base.json:61]: This needs to be added to cli.js.

File: oauth-proxy/dev-config.base.json [hunk 3]
```diff
@@ -57,6 +57,29 @@
...
+        "id": "0oaxxxlocalhostxxxxx"
+      },
+      {
+        "slug": "lighthouse",
+        "id": "0ua720axlighthousexxxxxxx"
+      },
+      {
+        "slug": "ssoi",
+        "id": "0uahh383xssoixxxxxx",
+        "manage": "some_manage_endpoint"
```
Comment (beaugrantham-va) [inline oauth-proxy/dev-config.base.json:73]: Let's remove the placeholder attributes until we actually define them.

File: oauth-proxy/dev-config.base.json [hunk 4]
```diff
@@ -57,6 +57,29 @@
...
-    }
+    },
+    "idps": [
+      {
+        "slug": "localhost",
+        "id": "0oaxxxlocalhostxxxxx"
+      },
+      {
+        "slug": "lighthouse",
+        "id": "0ua720axlighthousexxxxxxx"
```
Comment (beaugrantham-va) [inline oauth-proxy/dev-config.base.json:68]: Let's set the real ID here (pointing to dev) as it's not sensitive.

- [ ] Use real ID.

File: oauth-proxy/dev-config.base.json [hunk 5]
```diff
@@ -57,6 +57,29 @@
...
+      },
+      {
+        "slug": "ssoi",
+        "id": "0uahh383xssoixxxxxx",
```
Comment (beaugrantham-va) [inline oauth-proxy/dev-config.base.json:72]: Let's set the real ID here (pointing to dev) as it's not sensitive.

- [ ] Use real ID.

File: oauth-proxy/dev-config.base.json [hunk 6]
```diff
@@ -57,6 +57,29 @@
...
+        "manage": "some_manage_endpoint"
+      },
+      {
+        "slug": "ssoe",
```
Comment (beaugrantham-va) [inline oauth-proxy/dev-config.base.json:76]: We don't actually have an SSOe IDP yet, so let's remove this until it's ready.

- [ ] Remove SSOe.

**Review by beaugrantham-va — CHANGES_REQUESTED** [review]
Open devops (and oauth-proxy config repo) PR's to update real environment configs.

- [ ] Devops PR(s)
- [ ] oauth-proxy config repo PR(s)

**Comment by beaugrantham-va** [comment]
Looks good.

Good to merge once the devops (and oauth-proxy config repo) PRs are created. This can go ahead of them but I want to make sure those aren't forgotten.

**Comment by crolarlibertyva** [comment]
> Open devops (and oauth-proxy config repo) PR's to update real environment configs.
> 
>     * [ ]  Devops PR(s)
> 
>     * [ ]  oauth-proxy config repo PR(s)

https://github.com/department-of-veterans-affairs/devops/pull/8483

**Review by beaugrantham-va — APPROVED** [review]

**Pull request merged**